spec/concurrent/atomic/condition_spec.rb

```diff
@@ -32,37 +27,80 @@ module Concurrent
         describe '#wait without timeout' do
 
           it 'should block the thread' do
-            t = Thread.new { mutex.synchronize { subject.wait(mutex) } }
-            sleep(0.1)
-            expect(t.status).to eq 'sleep'
+            latch_1 = Concurrent::CountDownLatch.new
+            latch_2 = Concurrent::CountDownLatch.new
+            mutex = Mutex.new
+
+            t = Thread.new do
+              mutex.synchronize do
+                latch_1.count_down
+                subject.wait(mutex)
+                latch_2.count_down
+              end
+            end
+
+            latch_1.wait(1)
+            latch_2.wait(0.1)
```

Comment:
It might be better practice to use Events in similar cases giving the variable some name for improved readability. 
```ruby
  before_waiting = Concurrent::Event.new
  after_waiting = Concurrent::Event.new
  mutex = Mutex.new

  thread = Thread.new do
    mutex.synchronize do
      before_waiting.set
      subject.wait(mutex)
      after_waiting.set
    end
  end

  before_waiting.wait(1)
  after_waiting.wait(0.1)
  expect(after_waiting.set?).to be_false
  thread.kill
```
But both latch and event cannot be used here because both depend on the `Condition` which is being testes here. `Queue` may be good replacement here.